datamol/io.py

```diff
@@ -591,3 +592,107 @@ def to_xlsx(
 
     with fsspec.open(urlpath, mode="wb") as f:
         PandasTools.SaveXlsxFromFrame(mols, f, molCol=mol_column, size=mol_size)
+
+
+EXTENSIONS_DICT = {
+    "csv": [
+        ".csv",
+        ".csv.gz",
+        ".csv.bz2",
+        ".csv.zip",
+        ".csv.xz",
+        ".csv.zst",
+        ".csv.tar",
+        ".csv.tar.gz",
+        ".csv.tar.xz",
+        ".csv.tar.bz2",
+    ],
+    "excel": [".xlsx"],
+    "parquet": [".parquet"],
+    "json": [
+        ".json",
+        ".json.gz",
+        ".json.bz2",
+        ".json.zip",
+        ".json.xz",
+        ".json.zst",
+        ".json.tar",
+        ".json.tar.gz",
+        ".json.tar.xz",
+        ".json.tar.bz2",
+    ],
+    "sdf": [
+        ".sdf",
+        ".sdf.gz",
+    ],
+}
+
+
+def _guess_filetype(path: str):
+    """Return a filetype given an input path. Filetypes returned can be from
+    `csv, excel, parquet, json, sdf`.
+    """
+    for name, extensions in EXTENSIONS_DICT.items():
+        for ext in extensions:
+            if path.endswith(ext):
+                return name
+
+
+def open_df(path: str, **kwargs: Any) -> pd.DataFrame:
+    """Open a dataframe file whatever its filetype from
+    `csv, excel, parquet, json, sdf`.
+
+    Args:
+        path: path to the file.
+        **kwargs: keyword arguments to pass to the underlying reader.
+    """
+
+    filetype = _guess_filetype(path)
+
+    data = None
+    if filetype == "csv":
+        data = pd.read_csv(path, **kwargs)
+    elif filetype == "excel":
+        data = pd.read_excel(path, **kwargs)
+    elif filetype == "parquet":
+        data = pd.read_parquet(path, **kwargs)
+    elif filetype == "json":
+        data = pd.read_json(path, **kwargs)
+    elif filetype == "sdf":
+        data = dm.read_sdf(path, as_df=True, **kwargs)
+    else:
+        raise ValueError(f"The file type of {path} is not supported.")
+
+    data = cast(pd.DataFrame, data)
+
+    return data
+
+
+def save_df(
+    data: pd.DataFrame,
+    path: str,
+    **kwargs: Any,
+):
+    """Save a dataframe file whatever its filetype from
+    `csv, excel, parquet, json, sdf`.
+
+    Args:
+        data: dataframe to save.
+        path: path to save the file.
+        **kwargs: additional arguments to pass that are specific to the file save type.
+    """
+
+    filetype = _guess_filetype(path)
+
+    if filetype == "csv":
+        data.to_csv(path, index=False, **kwargs)
+    elif filetype == "excel":
+        data.to_excel(path, index=False, **kwargs)
```

Comment:
Sorry, I forgot to tell you earlier but with `index=False, **kwargs` it won't be possible to pass an `index=True` arg here.

Maybe use `kwargs.setdefault("index", False)` before `to_excel` and `to_csv`?